sqlmesh/utils/pydantic.py

```diff
@@ -309,10 +310,9 @@ def _get_fields(
     results = []
 
     for expr in expressions:
-        expr = normalize_identifiers(
-            exp.column(expr) if isinstance(expr, exp.Identifier) else expr,
-            dialect=dialect,
-        )
+        expr = exp.column(expr) if isinstance(expr, exp.Identifier) else expr
+        if dialect:
```

Comment:
what are the instances where there is or isn't a dialect?

Reply:
If the validator is used as part of the Model `kind` object the dialect won't be a part of the `values`. If it's used as par of the `ModelMeta` then it's available and can be used.

Reply:
is that ok? i haven't thought about that enough

Reply:
I'm also wondering if this is ok. Are you considering fixing that out of scope of this PR or not something that needs to be fixed?

Reply:
So it's only relevant when we deserialize kind from state. In which case it's fine to not have a dialect cause those values will be serialized without a dialect as well.